src/core/lib/gpr/tls.h

```diff
@@ -21,35 +21,36 @@
 
 #include <grpc/support/port_platform.h>
 
+#include <type_traits>
+
+template <typename T, typename = typename std::enable_if<(
+                          std::is_trivially_destructible<T>::value &&
+                          sizeof(T) <= sizeof(void*) &&
+                          alignof(void*) % alignof(T) == 0)>::type>
+class Wrapper {
```

Comment:
I know I typed Wrapper in the comment thread, but let's give it a more supportable name long term...

```
namespace grpc_core {
template <...> class TlsTypeWrapper { ... };
```

Reply:
Sure, done. Renamed the pthread type to look more C++ish as well.